### PR TITLE
TRUNK-5033 Public static fields should be final

### PR DESCRIPTION
**Pull request by dilekamadushan**
TRUNK-5033 Public static fields should be final

JIRA TICKET LINK- https://issues.openmrs.org/browse/TRUNK-5033?filter=-1

I modified public static variables into public static final

## Checklist: I completed these to help reviewers :)

 My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

 My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

 I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

 I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

 All new and existing **tests passed**.

 My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by dkayiwa** [comment]
Did you get a chance to look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dilekamadushan** [comment]
@dkayiwa Yeah I read it...Is there something wrong in my pull request ?

**Comment by dkayiwa** [comment]
For instance, that page talks of including the JIRA ticket link(url), which you have not done. Can you carefully read it again?

File: api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java [hunk 1]
```diff
@@ -12,25 +12,25 @@
...
 	
 	// 0 second delay added before the initial start of a task
-	public static long SCHEDULER_DEFAULT_DELAY = 0;
+	public static final long SCHEDULER_DEFAULT_DELAY = 0;
 	
-	public static String SCHEDULER_DEFAULT_USERNAME = "admin";
+	public static final String SCHEDULER_DEFAULT_USERNAME = "admin";
 	
-	public static String SCHEDULER_DEFAULT_PASSWORD = "test";
+	public static final String SCHEDULER_DEFAULT_PASSWORD = "test";
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java]: I think you might want to remove the code in SchedulerUtil.startup() that assigns values to these fields.

File: api/src/main/java/org/openmrs/util/Security.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
 	/**
 	 * encryption settings
 	 */
-	public static Logger log = LoggerFactory.getLogger(Security.class);
+	public static final Logger log = LoggerFactory.getLogger(Security.class);
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/util/Security.java]: I think we changed to not having loggers being static, can we clean it up here as well

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 3]
```diff
@@ -125,21 +125,21 @@ public static String getOpenmrsProperty(String property) {
 		return null;
 	}
 		
-	public static String DATABASE_NAME = "openmrs";
+	public static final String DATABASE_NAME = "openmrs";
 	
-	public static String DATABASE_BUSINESS_NAME = "openmrs";
+	public static final String DATABASE_BUSINESS_NAME = "openmrs";
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: The above 2 are also reassigned later

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 4]
```diff
@@ -125,21 +125,21 @@ public static String getOpenmrsProperty(String property) {
...
-	public static String OBSCURE_PATIENTS_GIVEN_NAME = "Demo";
+	public static final String OBSCURE_PATIENTS_GIVEN_NAME = "Demo";
 	
-	public static String OBSCURE_PATIENTS_MIDDLE_NAME = null;
-	
-	public static String OBSCURE_PATIENTS_FAMILY_NAME = "Person";
+	public static final String OBSCURE_PATIENTS_MIDDLE_NAME = null;
 	
+	public static final String OBSCURE_PATIENTS_FAMILY_NAME = "Person";
+
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: The above fields prefixed with OBSCURE_ are reassigned later

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 5]
```diff
@@ -42,7 +42,7 @@
 	
 	private static final Logger log = LoggerFactory.getLogger(OpenmrsConstants.class);
 	
-	public static String KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = "OPENMRS_APPLICATION_DATA_DIRECTORY";
+	public static final String KEY_OPENMRS_APPLICATION_DATA_DIRECTORY = "OPENMRS_APPLICATION_DATA_DIRECTORY";
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: This is reassigned later

File: api/src/main/java/org/openmrs/util/Security.java (hunk 2 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/util/Security.java]: @wluyima  Ok, When I modify it as public Logger log, it produces errors in other parts of the code with "non static fields can not be referenced in static methods".. Shall I go on and fix these errors as well ?

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java (hunk 4 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @wluyima  OpenmrsConstants.OBSCURE_PATIENTS is modified in PersonNameTest.java. Shall I remove those lines ?

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java (hunk 3 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @wluyima Sorry, I couldn't find a place where DATABASE_BUSINESS_NAME is modified

File: api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java (hunk 1 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java]: @wluyima on it :)

**Comment by dkayiwa** [comment]
@dilekamadushan did you see the merge conflicts?

File: api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java (hunk 1 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java]: @wluyima I removed that code and pushed..Thanks

**Comment by dilekamadushan** [comment]
@dkayiwa Yes, I am working on it. Thanks

File: api/src/main/java/org/openmrs/scheduler/SchedulerUtil.java [hunk 6]
```diff
@@ -30,20 +30,7 @@
 	 * @param p properties used to start the service
 	 */
 	public static void startup(Properties p) {
-		// Override the Scheduler constants if specified by the user
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/scheduler/SchedulerUtil.java]: @dilekamadushan please remove any change made to `SchedulerUtil` as @wluyima mentioned

File: api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java [hunk 7]
```diff
@@ -12,25 +12,25 @@
...
 	
 	/** The default 'from' address for emails send by the schedule */
-	public static final String SCHEDULER_DEFAULT_FROM = "scheduler@openmrs.org";
+  public static final String SCHEDULER_DEFAULT_FROM = "scheduler@openmrs.org";
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/scheduler/SchedulerConstants.java]: please fix the indentation (we use tabs)

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java (hunk 3 above)
Comment (teleivo) [reply api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @dilekamadushan you should also run `mvn clean package` before you push your changes for review. You should get compile errors if you make something a constant which turns out to be reassigned.

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 8]
```diff
@@ -418,63 +418,7 @@ public static boolean folderContains(File folder, String filename) {
 	 * @param p properties from runtime configuration
 	 */
 	public static void startup(Properties p) {
-		
-		// Override global OpenMRS constants if specified by the user
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: why did you remove the code here? it is still needed. You should not remove code so you can make a variable a constant. If you make a variable a constant and you get a compile error it just means that we are not able to make it a constant. The task of this ticket is simply to make things constants which are never reassigned.

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 9]
```diff
@@ -208,102 +208,7 @@ public void equalsContent_shouldReturnTrueIfOnlyInContentFieldsDifferenceIsBetwe
 		
 		assertThat(pn.equalsContent(other), is(true));
 	}
-	
-	/**
-	 * @see PersonName#getFamilyName()
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: this should also not be deleted

File: web/src/main/java/org/openmrs/web/Listener.java [hunk 10]
```diff
@@ -275,11 +275,7 @@ private void loadConstants(ServletContext servletContext) {
 		WebConstants.BUILD_TIMESTAMP = servletContext.getInitParameter("build.timestamp");
 		WebConstants.WEBAPP_NAME = getContextPath(servletContext);
 		WebConstants.MODULE_REPOSITORY_URL = servletContext.getInitParameter("module.repository.url");
-		
-		if (!"openmrs".equalsIgnoreCase(WebConstants.WEBAPP_NAME)) {
```
Comment (teleivo) [inline web/src/main/java/org/openmrs/web/Listener.java]: please dont remove this as well

File: api/src/main/java/org/openmrs/scheduler/SchedulerUtil.java (hunk 6 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/scheduler/SchedulerUtil.java]: @teleivo ok I will do the necessary modifications. Thanks

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java (hunk 8 above)
Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: @teleivo I will do the necessary changes. Thanks

**Comment by dilekamadushan** [comment]
@teleivo Thanks. I will focus on #2485.

**Comment by teleivo** [comment]
@dilekamadushan did you try `git pull --rebase upstream master` ?

**Comment by dilekamadushan** [comment]
@teleivo I tried it. I am getting coveralls errors in files I didn't edit. I referred to your git tutorials too. Please feel free to review. Thanks

**Comment by dilekamadushan** [comment]
@teleivo at 2017-12-29 01:58 GMT All the checks passed. Thanks

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 11]
```diff
@@ -39,9 +39,24 @@
...
  */
 public final class OpenmrsConstants {
-
-	private OpenmrsConstants() {
-	}
+	/**
+	 * The directory which OpenMRS should attempt to use as its application data directory
+	 * in case the current users home dir is not writeable (e.g. when using application servers
+	 * like tomcat to deploy OpenMRS).
+	 *
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: why are you moving this up? I think this section should stay where it was. try to change only whats necessary to achieve the goal of the issue you are working on

Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @teleivo ok I will make the necessary changes.

**Comment by dilekamadushan** [comment]
@teleivo I modified the code on your latest reviews. Feel free to review. Thanks

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 12]
```diff
@@ -154,22 +166,6 @@ public static String getOpenmrsProperty(String property) {
 	 * in case the current users home dir is not writeable (e.g. when using application servers
 	 * like tomcat to deploy OpenMRS).
 	 *
-	 * @see #APPLICATION_DATA_DIRECTORY_RUNTIME_PROPERTY
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: why did you move this part?

**Comment by dilekamadushan** [comment]
@teleivo I modified the code on your latest reviews. Feel free to review. Thanks

**Comment by dilekamadushan** [comment]
@teleivo I modified the code on your latest reviews. Please help me to identify what's wrong here. Thanks

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 13]
```diff
@@ -39,9 +39,20 @@
  * methods will be written to the database at startup if they don't exist yet.
  */
 public final class OpenmrsConstants {
-
-	private OpenmrsConstants() {
-	}
+	/**
+	 * @see OpenmrsUtil#getApplicationDataDirectory()
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: why did you move these `APPLICATION_DATA` constants up? I don't see the reason. Can you please undo this change.

Comment (dilekamadushan) [reply api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @teleivo ok I will make the necessary changes.Thanks

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 14]
```diff
@@ -40,8 +40,16 @@
  */
 public final class OpenmrsConstants {
 
-	private OpenmrsConstants() {
-	}
+	public static final String APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX = "/var/lib";
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @dilekamadushan its still the same

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 15]
```diff
@@ -39,9 +39,15 @@
  * methods will be written to the database at startup if they don't exist yet.
  */
 public final class OpenmrsConstants {
-
-	private OpenmrsConstants() {
-	}
+	
+	public static final String APPLICATION_DATA_DIRECTORY_FALLBACK_WIN = System.getenv("appdata");
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: @dilekamadushan why do you still move these `APPLICATION_DATA` constants?

**Comment by dilekamadushan** [comment]
@teleivo Should I create a new pull request ?, This branch seems closed.

**Comment by teleivo** [comment]
@dilekamadushan you closed this PR. you can either reopen or create a new one. what was the argument of codacy?

**Comment by dilekamadushan** [comment]
@teleivo There is no option to reopen the PR. And it displays Merged Successfully message as well. I am confused.

**Comment by teleivo** [comment]
@dilekamadushan it has not been merged. You can look at the commits. Please reopen another PR with your changes

**Comment by dilekamadushan** [comment]
@teleivo I find my changes been merged.I checked it now

**Comment by teleivo** [comment]
can you please paste the link to the merge commit I  do not see it

**Comment by dilekamadushan** [comment]
@teleivo You were right. I was mistaken, it has not been merged eventhough it is displayed. I will create a new PR. Thanks for the support all the way.
